docs/content/concepts/assets/asset-materializations.mdx

```diff
@@ -5,6 +5,8 @@ description: Dagster provides functionality to record and track changes to exter
 
 # Asset Materializations
 
+In most cases, you should be able to work with assets using <PyObject object="asset" decorator />, but there may be situations where you want to create an asset in the body of a <PyObject object="op" decorator />. You can do this using <PyObject object="AssetMaterialization" /> events.
```

Comment:
is this actually true??

Reply:
Instead of saying "in most cases" / "may be situations", I think we could say what the cases are.  E.g. something like "When it's not known what asset an op is going to materialize until the op runs..."

Separate thing - AssetMaterializations can also be recorded from IOManager `handle_output` functions.